src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -677,5 +493,128 @@ private void RemoveValueFromFile<T>(string fileName, string key)
+
+    /// <summary>
+    /// The GroupPolicy related settings used in PowerShell are as follows in Registry:
+    ///  - Software\Policies\Microsoft\Windows\PowerShell -- { EnableScripts (0 or 1); ExecutionPolicy (string) }
+    ///      SubKeys                  Name-Value-Pairs
+    ///      - ScriptBlockLogging     { EnableScriptBlockLogging (0 or 1); EnableScriptBlockInvocationLogging (0 or 1) }
+    ///      - ModuleLogging          { EnableModuleLogging (0 or 1); ModuleNames (string[]) }
+    ///      - Transcription          { EnableTranscripting (0 or 1); OutputDirectory (string); EnableInvocationHeader (0 or 1) }
+    ///      - UpdatableHelp          { DefaultSourcePath (string) }
+    ///      - ConsoleSessionConfiguration { EnableConsoleSessionConfiguration (0 or 1); ConsoleSessionConfigurationName (string) }
+    ///  - Software\Policies\Microsoft\Windows\EventLog
+    ///     SubKeys                   Name-Value-Pairs
+    ///      - ProtectedEventLogging  { EnableProtectedEventLogging (0 or 1); EncryptionCertificate (string[]) }
+    ///
+    /// The JSON representation is in sync with the 'PowerShellPolicies' type. Here is an example:
+    /// {
+    ///   "PowerShellPolicies": {
+    ///     "ScriptExecution": {
+    ///       "ExecutionPolicy": "RemoteSigned"
+    ///     },
+    ///     "ScriptBlockLogging": {
+    ///       "EnableScriptBlockInvocationLogging": true,
+    ///       "EnableScriptBlockLogging": false
+    ///     },
+    ///     "ProtectedEventLogging": {
+    ///       "EnableProtectedEventLogging": false,
+    ///       "EncryptionCertificate": [
+    ///         "Joe"
+    ///       ]
+    ///     },
+    ///     "Transcription": {
+    ///       "EnableTranscripting": true,
+    ///       "EnableInvocationHeader": true,
+    ///       "OutputDirectory": "c:\tmp"
+    ///     },
+    ///     "UpdatableHelp": {
+    ///       "DefaultSourcePath": "f:\temp"
+    ///     },
+    ///     "ConsoleSessionConfiguration": {
+    ///       "EnableConsoleSessionConfiguration": true,
+    ///       "ConsoleSessionConfigurationName": "name"
+    ///     }
+    ///   }
+    /// }
+    /// </summary>
+    internal sealed class PowerShellPolicies
+    {
+        public ScriptExecution ScriptExecution { get; set; }
+        public ScriptBlockLogging ScriptBlockLogging { get; set; }
+        public ModuleLogging ModuleLogging { get; set; }
+        public ProtectedEventLogging ProtectedEventLogging { get; set; }
+        public Transcription Transcription { get; set; }
+        public UpdatableHelp UpdatableHelp { get; set; }
+        public ConsoleSessionConfiguration ConsoleSessionConfiguration { get; set; }
+    }
+
+    internal abstract class PolicyBase { }
+
+    /// <summary>
+    /// Setting about ScriptExecution
+    /// </summary>
+    internal sealed class ScriptExecution : PolicyBase
+    {
+        public string ExecutionPolicy { get; set; }
+        public bool? EnableScripts { get; set; }
```

Comment:
Not sure I understand the value of having this nullable.  EnableScripts == null is same as EnableScripts == true, correct?  Same question for the nullable properties below where they should have a default value and being null is the same as the default value.

Reply:
`EnableScripts == null` means this setting is not defined in the config file, so it's up to powershell to define the default behavior. `EnableScripts == true` means it's defined in the config file and the value is `true`. Likewise, `EnableScripts == false` means it's defined in the config file and the value is `false`.

It's the same in registry -- a key is not defined, which means the application needs to use the pre-defined behavior; a key is defined, and the value is 1 or 0.

Reply:
An example is the use of `EnableScriptBlockLogging` in `LogScriptBlockCreation(ScriptBlock scriptBlock, bool force)`
- when it's explicitly set to 'true', we always do script block logging; 
- when it's not defined, we do script block logging when `force == true`
- when it's explicitly set to 'false', we always disable script block logging, even if `force == true`.

Reply:
Ok, the force case makes sense that there is a difference with null.  Thanks.

Reply:
I don't like the nullable too. I see no reason to emulate registry keys. If now we use static classes it is more readable and maintainable to set defaults *explicitly*. `Force` works with this too.

Reply:
I don't think it's mimicking registry keys, but a requirement to represent three states: enabled, disabled and not defined.